GitUI/CommitInfo/CommitInfo.cs

```diff
@@ -809,10 +809,12 @@ internal sealed class BranchComparer : IComparer<string>
                 RegexOptions.Compiled | RegexOptions.CultureInvariant);
 
             private readonly string _currentBranch;
+            private readonly bool _isDetachedHead;
 
             public BranchComparer(string currentBranch)
             {
                 _currentBranch = currentBranch;
+                _isDetachedHead = DetachedHeadParser.IsDetachedHead(currentBranch);
```

Comment:
Side note: DetachedHeadParser.IsDetachedHead() checks for specific patterns, all patterns with ( or space should be considered invalid and handled as detached head.
But that is separate